File: Selenium/Project/Activity6.py
"""Menu checking 
Goal: Make sure that the “Activities” menu item exists and is clickable 
a. Open a browser. 
b. Navigate to ‘http://alchemy.hguy.co/crm’ and login using the credentials 
provided. 
c. Locate the navigation menu. 
d. Ensure that the “Activities” menu item exists. 
e. Close the browser """

#Import the driver
from selenium import webdriver
from selenium.webdriver.common.by import By

#Initialize the firefox driver
with webdriver.Firefox() as driver:
    #Open the browser
    driver.get("http://alchemy.hguy.co/crm")
    #Find the username & password and login using the credentials
    username = driver.find_element(By.NAME, "user_name")
    password = driver.find_element(By.NAME, "username_password")
    username.send_keys("admin")
    password.send_keys("pa$$w0rd")
    driver.find_element(By.NAME, "Login").click()
    navigationMenu = driver.find_element(By.ID, "toolbar")
    print("Verify if navigation menu is visible: ", navigationMenu.is_displayed())
    Activities = driver.find_element(By.XPATH, "//a[contains(@id, 'grouptab_3')]")
    print("Verify if activities menu is displayed: ", Activities.is_displayed())
    Activities.click()
    print("Activity menu is clickable & drop down values are displayed in UI screen")

File: Selenium/Project/Activity6.java
/*
  a. Open a browser.
  b. Navigate to ‘http://alchemy.hguy.co/crm’ and login using the credentials provided.
  c. Locate the navigation menu. 
  d. Ensure that the “Activities” menu item exists. 
  e. Close the browser  
 */
package crmactivities;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;
import org.testng.annotations.AfterClass;

public class Activity6 {

	WebDriver driver;
	WebDriverWait wait;
	
	@BeforeClass
	public void setUp() {
		
		//Driver initialization
		driver = new FirefoxDriver();
		
		//Open the url
		driver.get("http://alchemy.hguy.co/crm");
	}
	
	@Test
	public void login() {
		
		WebElement userName = driver.findElement(By.name("user_name"));
		WebElement password = driver.findElement(By.name("username_password"));
		
		//Enter login credential
		userName.sendKeys("admin");
		password.sendKeys("pa$$w0rd");
		driver.findElement(By.name("Login")).click();
	}
	
	@Test
	public void navigationMenu() {
		
		WebElement crmMenu = driver.findElement(By.xpath("//div[@id = 'toolbar']"));
		System.out.println("Is navigation menu displayed - " + crmMenu.isDisplayed());
		WebElement activities = driver.findElement(By.xpath("//a[contains(@id, 'grouptab_3')]"));
		System.out.println("Is activities menu clickable - " + activities.isDisplayed());
		activities.click();
		System.out.println("Activity menu is clickable & dropdown lists are shown in the UI screen");
		wait = new WebDriverWait(driver, Duration.ofSeconds(10));
	}

	@AfterClass
	public void tearDown() {
		driver.quit();
	}
}
